### PR TITLE
Support for ordered categoricals.

### PR DESCRIPTION
**Pull request by ivirshup**
* IO supported via attribute on categories
* Tests now check order of categoricals

Were there other reasons we didn't support these other than having an on-disk representation?

### CODE REVIEW

**Comment by LuckyMD** [comment]
There were some issues in between with all categoricals being ordered. That created problems for integrating other DE methods with scanpy, as the DE methods treat ordered categoricals differently to unordered categoricals. I wonder if people may accidentally store categoricals as ordered and then get strange results in DE testing if this is supported. What do you need ordered categoricals for in sc analysis?

File: anndata/tests/helpers.py [hunk 1]
```diff
@@ -299,11 +300,6 @@ def are_equal_dataframe(a, b, exact=False, elem_name=None):
         b,
         check_index_type=exact,
         check_exact=exact,
-        # If False: check encoded values, but not codes
-        check_categorical=exact,
-        # Should different orderings be allowed?
-        # check_like=exact,
-        check_names=False,
```
Comment (flying-sheep) [inline anndata/tests/helpers.py]: @ivirshup can you check if my merge is correct? I simplified this code simulataneously to you changing it.

**Comment by ivirshup** [comment]
@LuckyMD, this doesn't remove support for unordered categorical or change the behaviour for automatic categorical conversion. Now it just means that if you do something like:

```python
adata.obs["time"] = pd.Categorical(times, categories=["1hr", "5hr", "10hr"], ordered=True)
```

That will be maintained on roundtrips to disk, and will be available for downstream use by libraries like `seaborn`. This is mostly inspired by some figures I've seen at talks where people "faked" ordered categories with `scanpy` by using numerical values and a continuous colormap. Real ugly stuff.

File: anndata/tests/helpers.py (hunk 1 above)
Comment (ivirshup) [reply anndata/tests/helpers.py]: Looks fine to me!

**Comment by LuckyMD** [comment]
I guess seaborn just orders the legend differently if you pass ordered categoricals? Or is there another effect?

**Comment by ivirshup** [comment]
Looks like it's now running into this https://github.com/linnarsson-lab/loompy/issues/113

**Comment by ivirshup** [comment]
Is anyone strongly against merging this?

**Comment by flying-sheep** [comment]
I’m happy with this!

**Pull request merged**